containers/src/Data/Tree.hs

```diff
@@ -532,6 +543,243 @@ unfoldForestQ f aQ = case viewl aQ of
         q' :> a -> splitOnto (a:as) bs q'
         EmptyR -> error "unfoldForestQ"
 
+-- | \(O(n)\). The leaves of the tree in left-to-right order.
+--
+-- A leaf is a node with no children.
+--
+-- ==== __Examples__
+--
+-- >>> leaves (Node 1 [Node 2 [Node 4 [], Node 5 []], Node 3 [Node 6 []]])
+-- [4,5,6]
+-- >>> leaves (Node "root" [])
+-- ["root"]
+--
+-- @since FIXME
+leaves :: Tree a -> [a]
+#ifdef __GLASGOW_HASKELL__
+leaves t = GHC.Exts.build $ \cons nil ->
+  let go (Node x []) z = cons x z
+      go (Node _ ts) z = foldr go z ts
+  in go t nil
+{-# INLINE leaves #-} -- Inline for list fusion
+#else
+leaves t =
+  let go (Node x []) z = x:z
+      go (Node _ ts) z = foldr go z ts
+  in go t []
+#endif
+
+-- | \(O(n)\). The edges of the tree as parent-child pairs in pre-order.
+--
+-- A tree with \(n\) nodes has \(n-1\) edges.
+--
+-- ==== __Examples__
+--
+-- >>> edges (Node 1 [Node 2 [Node 4 [], Node 5 []], Node 3 [Node 6 []]])
+-- [(1,2),(2,4),(2,5),(1,3),(3,6)]
+-- >>> edges (Node "root" [])
+-- []
+--
+-- @since FIXME
```

Comment:
I think it would be useful to indent the documentation to show tree structure.

Reply:
That would be nice, but it looks like Haddock doesn't support multi-line REPL examples. So we would have to give that up and use an ordinary code block. Would you still prefer that?

Reply:
I realized we can just pretend that Haddock supports it 😄 
Doesn't look too bad.
![screenshot-leaves](https://github.com/user-attachments/assets/1d9fcc98-1cbb-4793-9c50-dd5b0cb2a0fa)